partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java

```diff
@@ -125,33 +129,72 @@ private static String handlePartialUpdateForClassOrInterfaceFile(String generate
                 for (BodyDeclaration<?> generatedMember : generatedFileMembers) {
                     if (isMembersCorresponding(existingMember, generatedMember)) {
                         updatedMembersList.add(generatedMember);
+                        break;
                     }
                 }
             }
         }
 
-        // 5. add remaining members in generated file to the new members list
+        // 6. Add remaining members in generated file to the new members list
         for (BodyDeclaration<?> generatedMember : generatedFileMembers) {
-            boolean hasMembersWithSameName = false;
+            boolean needToAddToUpdateMembersList = true;
             for (BodyDeclaration<?> existingMember : updatedMembersList) {
-                if (isMembersWithSameName(existingMember, generatedMember)) {
-                    hasMembersWithSameName = true;
+                // If the generated member and the existing member is corresponding,
+                // or if there is an existing member who has the same name as the generated member and is manually written,
+                // Then we don't put the generated member to the updatedMembersList.
+                if (isMembersCorresponding(existingMember, generatedMember) || (isMembersWithSameName(existingMember, generatedMember) && !hasGeneratedAnnotation(existingMember))) {
+                    needToAddToUpdateMembersList = false;
+                    break;
                 }
             }
-            if (!hasMembersWithSameName) {
+            if (needToAddToUpdateMembersList) {
                 updatedMembersList.add(generatedMember);
             }
         }
 
-        // 6. update members
+        // 7. Update members
         generatedClazz.setMembers(updatedMembersList);
 
-        // 7. update imports
+        // 8. Update imports
         compilationUnitForGeneratedFile.getImports().addAll(compilationUnitForExistingFile.getImports());
 
         return compilationUnitForGeneratedFile.toString();
     }
 
+    /**
+     * Verify if the generatedFile is valid
+     * @param generatedFileMembers, members in the generated file
+     * @return true if the generated file is valid, otherwise return false
+     */
+    private static void validateGeneratedFile(List<BodyDeclaration<?>> generatedFileMembers) {
+        // 1. Verify there is no duplicate methods (methods with same signature are considered duplicate methods)
+        NodeList<CallableDeclaration<?>> methodList = new NodeList<>();
+        for (BodyDeclaration<?> generatedMember : generatedFileMembers) {
+            if (generatedMember.isCallableDeclaration()) {
+                if (isMethodExistsInMethodList(generatedMember.asCallableDeclaration(), methodList)) {
+                    throw new RuntimeException(String.format("Found duplicate methods in the generated file."));
+                }
+                methodList.add(generatedMember.asCallableDeclaration());
+            }
+        }
```

Comment:
there should be more efficient way to do this
1. use a Set<String> for signatures
2. use a List<String> for signatures, and sort and check duplicate